src/main/java/uk/gov/hmcts/reform/hmc/service/HearingActualsServiceImpl.java

```diff
@@ -0,0 +1,105 @@
+package uk.gov.hmcts.reform.hmc.service;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.hmc.data.HearingEntity;
+import uk.gov.hmcts.reform.hmc.exceptions.BadRequestException;
+import uk.gov.hmcts.reform.hmc.exceptions.HearingNotFoundException;
+import uk.gov.hmcts.reform.hmc.model.ActualHearingDay;
+import uk.gov.hmcts.reform.hmc.model.HearingActual;
+import uk.gov.hmcts.reform.hmc.repository.HearingRepository;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static uk.gov.hmcts.reform.hmc.exceptions.ValidationError.HEARING_ACTUALS_HEARING_DAYS_INVALID;
+import static uk.gov.hmcts.reform.hmc.exceptions.ValidationError.HEARING_ACTUALS_ID_NOT_FOUND;
+import static uk.gov.hmcts.reform.hmc.exceptions.ValidationError.HEARING_ACTUALS_INVALID_STATUS;
+import static uk.gov.hmcts.reform.hmc.exceptions.ValidationError.HEARING_ACTUALS_NON_UNIQUE_HEARING_DAYS;
+
+@Service
+public class HearingActualsServiceImpl implements HearingActualsService {
+    private final HearingHelper hearingHelper;
+    private final HearingRepository hearingRepository;
+
+    private static final List<String> ALLOWED_ACTUALS_STATUSES = List.of("LISTED",
+                                                                         "UPDATE_REQUESTED",
+                                                                         "UPDATE_SUBMITTED");
+    public static final List<String> HEARING_RESULTS_THAT_NEED_REASON_TYPE = List.of("ADJOURNED", "CANCELLED");
+
+    @Autowired
+    public HearingActualsServiceImpl(HearingRepository hearingRepository, HearingHelper hearingHelper) {
+        this.hearingRepository = hearingRepository;
+        this.hearingHelper = hearingHelper;
+    }
+
+    public void updateHearingActuals(Long hearingId, HearingActual request) {
+        hearingHelper.isValidFormat(hearingId.toString());
+        HearingEntity hearing = getHearing(hearingId);
+        String hearingStatus = hearing.getStatus();
+        validateHearingStatusForActuals(hearingStatus);
+        validateRequestPayload(request, hearing);
+
+    }
+
+    private void validateRequestPayload(HearingActual request, HearingEntity hearing) {
+        validateHearingActualDaysNotInTheFuture(request);
+        validateDuplicateHearingActualDays(request);
+        validateHearingActualDaysNotBeforeFirstHearingDate(request, hearing);
+        validateHearingResult(request);
+    }
+
+    private void validateHearingResult(HearingActual request) {
+        if (HEARING_RESULTS_THAT_NEED_REASON_TYPE.contains(request.getHearingOutcome().getHearingResult().toUpperCase())
+            && StringUtils.isBlank(request.getHearingOutcome().getHearingResultReasonType())) {
+            throw new BadRequestException(null);
```

Comment:
why null message

Reply:
There was no message defined in the AC. But after a chat with the BA we have decided to put one in here.